File: Producto.java
public class Producto {
    public static int CANTIDAD_PRODUCTOS = 0;

    private String nombre;
    private double precio;
    private String fechaImportacion;
    private int numeroSerie;
    private int stock;

    public Producto(String nombre, double precio, String fechaImportacion, int stock) {
        this.nombre = nombre;
        this.precio = precio;
        this.fechaImportacion = fechaImportacion;
        this.stock = stock;
        this.numeroSerie = CANTIDAD_PRODUCTOS += 1;
    }

    public void agregarStock(int cantidad) {
        stock += cantidad;
    }

    public void eliminarStock(int cantidad) {
        if (stock >= cantidad) {
            stock -= cantidad;
        } else {
            System.out.println("No hay suficiente stock.");
        }
    }

    private boolean validarStock() {
        return stock == 0;
    }

    public String obtenerInformacion() {
        return String.format("Nombre: %s, precio: %f, fecha importacion: %s, stock: %d, numero de serie: %d", nombre, precio, fechaImportacion, stock, numeroSerie);
    }

    public String getNombre() {
        return nombre;
    }

    public double getPrecio() {
        return precio;
    }

    public int getStock() {
        return stock;
    }
}

File: Producto.py
class Producto:
    CANTIDAD_PRODUCTOS = 0

    def __init__(self, nombre, precio, fecha_importacion, stock):
        self.nombre = nombre
        self.precio = precio
        self.fecha_importacion = fecha_importacion
        self.stock = stock
        self.numero_serie = Producto.CANTIDAD_PRODUCTOS + 1
        Producto.CANTIDAD_PRODUCTOS += 1

    def agregar_stock(self, cantidad):
        self.stock += cantidad

    def eliminar_stock(self, cantidad):
        if self.validar_stock():
            print("")
        else:
            self.stock -= cantidad

    def validar_stock(self):
        return self.stock == 0

    def obtener_informacion(self):
        return f"Nombre: {self.nombre}, Precio: {self.precio}, Fecha Importación: {self.fecha_importacion}, Stock: {self.stock}, Número de Serie: {self.numero_serie}"

    def get_nombre(self):
        return self.nombre

    def get_precio(self):
        return self.precio

    def get_stock(self):
        return self.stock
